base_layer/core/src/proof_of_work/difficulty.rs

```diff
@@ -148,4 +149,25 @@ mod test {
         let d = Difficulty::from(1_000_000);
         assert_eq!("1,000,000", format!("{}", d));
     }
+
+    #[test]
+    fn high_target() {
+        let target: &[u8] = &[0xff, 0xff, 0xff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0 ];
+        let expected = Difficulty::from(1);
+        assert_eq!(big_endian_difficulty(target), expected);
+    } 
+
+    #[test]
+    fn max_difficulty() {
+        let target = u64::MAX;
```

Comment:
This is actually the same test case as the one below. What you want here is a test that captures the path where the result in L#121 is equal to u64::MAX, such that the min function doesn't actually apply.

This should suffice:


            let target = U256::MAX / U256::from(u64::MAX);
            let mut bytes = [0u8; 32];
            target.to_big_endian(&mut bytes);
            assert_eq!(big_endian_difficulty(&bytes), Difficulty::from(u64::MAX));

Reply:
Good point, I understand the logic and refactored the code accordingly. In this case, L#122 computes the min between the two values of `u64::MAX`, as represented by type `U256`, am I correct? 

If I am not missing something, we now have test coverage for each possible situation regarding the method `big_endian_difficulty`.